### PR TITLE
nydus-image：optimize chunkdict save

### PR DESCRIPTION
**Pull request by cslinwang**
## Relevant Issue (if applicable)
_If there are Issues related to this PullRequest, please list it._

## Details
Refactor the Deduplicate implementation to only
initialize config when inserting chunk data.
Simplify code for better maintainability.

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: src/bin/nydus-image/deduplicate.rs [hunk 1]
```diff
@@ -195,7 +204,11 @@ impl Deduplicate<SqliteDatabase> {
             }
             Ok(())
         };
-        let tree = Tree::from_bootstrap(&self.sb, &mut ())
+        let sb = self
```
Comment (imeoer) [inline src/bin/nydus-image/deduplicate.rs:207]: How about `fn insert_chunks(&mut self, sb: &RafsSuper) -> anyhow::Result<()> {`

Comment (cslinwang) [reply src/bin/nydus-image/deduplicate.rs:207]: OK

**Review by imeoer — APPROVED** [review]

File: src/bin/nydus-image/main.rs [hunk 2]
```diff
@@ -399,7 +398,7 @@ fn prepare_cmd_args(bti_string: &'static str) -> App {
                     )
                     .arg(arg_output_json.clone())
             )
-    );
+                );
```
Comment (An-DJ) [inline src/bin/nydus-image/main.rs:401]: Can't believe that cargo fmt didn't find this...

Comment (imeoer) [reply src/bin/nydus-image/main.rs:401]: It seems `cargo fmt` is broken from L164. 😂

**Pull request merged**